src/wallet/wallet.cpp

```diff
@@ -1468,12 +1468,30 @@ void CWallet::transactionRemovedFromMempool(const CTransactionRef& tx, MemPoolRe
 
 void CWallet::blockConnected(ChainstateRole role, const interfaces::BlockInfo& block)
 {
-    if (role == ChainstateRole::BACKGROUND) {
-        return;
-    }
     assert(block.data);
     LOCK(cs_wallet);
 
+    switch (role) {
+        case ChainstateRole::BACKGROUND:
+            m_background_validation_height = block.height;
+            return;
+        case ChainstateRole::ASSUMEDVALID:
+            if (m_background_validation_height == -1) {
+                m_background_validation_height = 0;
+            }
+            break;
+        case ChainstateRole::NORMAL:
+            m_background_validation_height = -1;
+            break;
+    } // no default case, so the compiler can warn about missing cases
+
+    if (role == ChainstateRole::BACKGROUND) {
+        m_background_validation_height = block.height;
+        return;
+    } else if (role == ChainstateRole::NORMAL) {
+        m_background_validation_height = -1;
+    } else {}
```

Comment:
Conditions were covered by the `switch` above.
```suggestion
```

Reply:
Dropped.